examples/2d/rotation.rs

```diff
@@ -0,0 +1,253 @@
+/// * If the result is negative the vectors are pointing in roughly opposite directions, the angle
+/// between them is greater than 90 and less than 180 degrees.
+///
+/// It is possible to get the angle by taking the arc cosine (`acos`) of the dot product. It is
+/// often unnecessary to do this though. Beware than `acos` will return `NaN` if the input is less
+/// than -1.0 or greater than 1.0. This can happen even when working with unit vectors due to
+/// floating point precision loss, so it pays to clamp your dot product value before calling
+/// `acos`.
+fn rotate_to_player_system(
+    mut query: Query<(&RotateToPlayer, &mut Transform), Without<Player>>,
+    player_query: Query<&Transform, With<Player>>,
+) {
+    let player_transform = player_query.single();
+    // get the player translation in 2d
+    let player_translation = player_transform.translation.xy();
+
+    for (config, mut enemy_transform) in query.iter_mut() {
+        // get the enemy ship forward vector in 2d (already unit length)
+        let enemy_forward = (enemy_transform.rotation * Vec3::Y).xy();
+
+        // get the vector from the enemy ship to the player ship in 2d and normalize it.
+        let to_player = (player_translation - enemy_transform.translation.xy()).normalize();
+
+        // get the dot product between the enemy forward vector and the direction to the player.
+        let forward_dot_player = enemy_forward.dot(to_player);
+
+        // if the dot product is approximately 1.0 then the enemy is already facing the player and
+        // we can early out.
+        if (forward_dot_player - 1.0).abs() < f32::EPSILON {
+            continue;
+        }
+
+        // get the right vector of the enemy ship in 2d (already unit length)
+        let enemy_right = (enemy_transform.rotation * Vec3::X).xy();
+
+        // get the dot product of the enemy right vector and the direction to the player ship.
+        // if the dot product is negative them we need to rotate counter clockwise, if it is
+        // positive we need to rotate clockwise. Note that `copysign` will still return 1.0 if the
+        // dot product is 0.0 (because the player is directly behind the enemy, so perpendicular
+        // with the right vector).
+        let right_dot_player = enemy_right.dot(to_player);
+
+        // determine the sign of rotation from the right dot player. We need to negate the sign
+        // here as the 2D bevy co-ordinate system rotates around +Z, which is pointing out of the
+        // screen. Due to the right hand rule, positive rotation around +Z is counter clockwise and
+        // negative is clockwise.
+        let rotation_sign = -f32::copysign(1.0, right_dot_player);
+
+        // limit rotation so we don't overshoot the target. We need to convert our dot product to
+        // an angle here so we can get an angle of rotation to clamp against.
+        let max_angle = forward_dot_player.min(1.0).max(-1.0).acos(); // clamp acos for safety
```

Comment:
```suggestion
        let max_angle = forward_dot_player.clamp(-1.0, 1.0).acos(); // clamp acos for safety
```

Reply:
Sure, I was intentionally avoiding clamp because it asserts that the min value is less than the max value, which is unnecessary here, but it's probably not worth worrying about.